ci_framework/plugins/action/discover_latest_image.py

```diff
@@ -39,67 +40,58 @@
 
 from ansible.errors import AnsibleError
 from ansible.plugins.action import ActionBase
-from ansible.module_utils.basic import missing_required_lib
 
 import re
 
-try:
-    from bs4 import BeautifulSoup
-except ImportError as imp_exc:
-    BS4_IMPORT = imp_exc
-else:
-    BS4_IMPORT = None
-
 
 class ActionModule(ActionBase):
     def run(self, tmp=None, task_vars=None):
         super(ActionModule, self).run(tmp, task_vars)
         module_args = self._task.args.copy()
 
-        if BS4_IMPORT:
-            raise AnsibleError(missing_required_lib('beautifulsoup4'))
-
         if 'image_prefix' not in module_args:
             raise AnsibleError('"image_prefix" parameter is mandatory')
 
+        if 'images_file' not in module_args:
```

Comment:
Why not just defaulting here? At the module level?

Reply:
I added it in `defaults`. But I think having this check isn't a bad thing.